src/utils.jl

```diff
@@ -205,6 +211,32 @@ initial_concentrations(
     ) for (k, s) in m.species
 )
 
+"""
+    isfreein(id::String, expr::SBML.Math)
+
+Determine if `id` is used and not bound (aka. free) in `expr`.
+"""
+isfreein(id::String, expr::SBML.Math) = interpret_math(
+    expr,
+    map_apply = any(rec.(x.body)),
+    map_const = _ -> false,
+    map_ident = x -> x.id == id,
+    map_lambda = (x, rec) -> id in x.args ? false : rec(x.body),
+    map_time = _ -> false,
+    map_value = _ -> false,
+)
+
+"""
+    isboundbyrules(
+        id::String,
+        m::SBML.Model
+    )
+
+Determine if an identifier is defined or used by any Rules in the model.
+"""
+isboundbyrules(id::String, m::SBML.Model) =
+    any(r.id == id for r in m.rules if typeof(r) in [SBML.AssignmentRule, SBML.RateRule]) ||
+    any(isfreein(Ref(id), r.math for r in m.rules))
```

Comment:
I thought this needs to be `any(isfreein(Ref(id), r.math for r in m.rules if r isa SBML.AlgebraicRule))`. Cause if say a compartment appears on the rhs of an Assignment or RateRule it is not bound by that rule. Only if it appears on the lhs it is (which is exactly what is checked in the line above).

Reply:
I see @exaexa . I mean we never divide by a Float in extensive_kinetic_math. Either we divide by compartment.size or we return `species123` without division, which obviously is the same as dividing by 1.0.
But we only ignore compartment size if `isnothing(compartment.size) && !(compartment.spatial_dimensions == 0)`. However, in these cases `rateof(somethingElse) = somefunction ( compartment123 )` would have to use a `nothing` sized compartment and the whole simulation would fail there. And tbh, I think it should. A `nothing` sized compartment should not appear on the rhs of a Rate or AssignmentRule.

Reply:
Ah yes that probably corresponds to my other question below, I'd say it's cleared now if what I assumed there is right